go.mod

```diff
@@ -108,5 +108,7 @@ exclude google.golang.org/grpc/examples v0.0.0-20200728065043-dfc0c05b2da9
 // Used for /-/reload
 replace github.com/weaveworks/common => github.com/rfratto/weaveworks-common v0.0.0-20210326192855-c95210d58ba7
 
-// loadbalancingexporter uses non-fixed version of batchpertrace which fetches latest and causes problems
-replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpertrace => github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpertrace v0.21.0
+// loadbalancingexporter uses non-fixed version of batchpersignal which fetches latest and causes problems
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpersignal => github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpersignal v0.27.0
+
+replace go.opentelemetry.io/collector => github.com/mapno/opentelemetry-collector v0.24.1-0.20210610132247-eb06509beaa2
```

Comment:
Can you move this to the grafana org? IMO we should have as few personal forks as possible. (I'm the worst offender of this and need to remove the rfratto instances here)

Reply:
Any particular reason we're not vendoring otelcol directly?

Reply:
> What does v0.24.1-... refer to? Is it this commit: mapno/opentelemetry-collector@eb06509 Those Go version always confuse me.

Correct. It's referencing that commit. I also find go modules confusing, but the reason for that _version_ is that it's referencing an unreleased commit, thus it's using a [pseudo-version](https://golang.org/ref/mod#pseudo-versions). If I'm not mistaken, it's using the 3rd form for pseudo-versions, but I don't know why it's v0.24.x and not v0.27.x.

> Are you planning to PR these changes? If so, we can link the PR as comment, makes it easier to track when we can remove the fork.

> Any particular reason we're not vendoring otelcol directly?

Yes, that's the plan. I've just opened the [PR](https://github.com/open-telemetry/opentelemetry-collector/pull/3418) to the collector. Initially I used a fork to get things moving fast, so we could integrate rapidly the new sampling policies we're working on. It'll probably won't be necessary to use my fork or Grafana's fork if it gets merged soon.

> Can you move this to the grafana org?

I'll wait a bit to get a response from the upstream. If it doesn't get merged we can create the org fork and make it _official_.

Reply:
:woohoo: merged! https://github.com/open-telemetry/opentelemetry-collector/pull/3418